enhancements/cluster-logging/cluster-logging-v2-apis.md

```diff
@@ -0,0 +1,514 @@
+
+The **cluster-observability-operator**:
+1. Deploys the logging-view-plugin for reading logs in the OpenShift console
+
+The **loki-operator**:
+1. Deploys the **LokiStack** for storing logs on-cluster
+
+The **cluster-logging-operator**:
+
+1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace
+
+### API Extensions
+
+This API defines the following opinionated input sources which is a continuation of prior cluster logging versions:
+
+* **application**: Logs of container workloads running in all namespaces except **default**, **openshift***, and **kube***
+* **infrastructure**: journald logs from OpenShift nodes and container workloads running only in namespaces **default**, **openshift***, and **kube***
+* **audit**: The logs from OpenShift nodes written to the node filesystem by: Kubernetes API server, OpenShift API server, Auditd, and OpenShift Virtual Network (OVN).
+
+These are **well-known** input sources that can be referenced by a pipeline without an explicit input specification.
+
+More explicit specification of **audit** and **infrastructure** logs is allowed by creating a named input of that type and specifiying at least one of the allowed sources.
+
+#### CluserLogForwarer CustomResourceDefinition:
+
+Following defines the next version of a ClusterLogForwarder. **Note:** The next version of this resources is part of a new API group to align log collection with
+the objectives of Red Hat observability.
+
+```yaml
+    apiVersion: "observability.openshift.io/v1"
+    kind: ClusterLogForwarder
+    metadata:
+      name:
+    spec:
+      managementState:   #enum: Managed, Unmanaged
+      serviceAccount:
+        name:
+      collector:
+        resources:       #corev1.ResourceRequirements
+          limits:        #cpu, memory
+          requests:
+        nodeSelector:    #map[string]string
+        tolerations:     #corev1.Toleration
+      inputs:
+      - name:
+        type:                      #enum: application,infrastructure,audit
+        application:
+          selector:                #labelselector
+          includes:
+          - namespace:
+            container:
+          excludes:
+          - namespace:
+            container:
+          tuning:
+            ratelimitByContainer:  #rate limit applied to each container selected by this input
+              recordsPerSecond:    #int  (no multiplier, a each container only runs on one node at a time.)
+        infrastructure:
+          sources: []              #enum: node,container
+        audit:
+          sources: []              #enum: auditd,kubeAPI,openshiftAPI,ovn
+        receiver:
+          type:                    #enum: syslog,http
+      filters:
+      - name:
+        type:                      #enum: kubeapiaudit, detectmultiline, parse, labels, drop, prune
+        kubeAPIAudit:
+        parse:
+      pipelines:
+       - inputRefs: []
+         outputRefs: []
+         filterRefs: []
```

Comment:
> We have (I have) previously talked about splitting the definitions of inputs, outputs and filters into separate resources, so that they can potentially be re-used for multiple CLFs.
> 
> Did you already explore this idea any further? 
We have not but I believe given this proposal was not initiated several months earlier, there isn't time to add a net-new concept and implement it.  It probably would also be best, if we wanted, to make it part of a beta api.

>Would this make sense in our context or do we usually have a 1:1 relationship between forwarders and their inputs/outputs?

Prior to 5.9, their was limited mechanism to support wiring a 1:1 and the implementation did not result in a dedicated pipeline either.  The end result was the backpressure from one output could negatively affect all outputs.   I don't believe we have enough evidence yet how users are going to use the changes from 5.9.  Additionally, I don't imagine we have robust telemetry that will tell us this information either.  I can imagine potentially the re-use of outputs but given we don't currently intend "normal" users to create CLFs, in theory the number of forwarders should be minimal